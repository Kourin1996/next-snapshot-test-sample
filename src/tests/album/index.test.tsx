import { getPage } from "next-page-tester";
import fetchMock from "fetch-mock";
import renderer from "react-test-renderer";

describe("album page", () => {
  beforeAll(() => {
    fetchMock
      .get("https://jsonplaceholder.typicode.com/albums/1", {
        id: 1,
        userId: 1,
        title: "Test Title",
      })
      .get("https://jsonplaceholder.typicode.com/albums/1/photos", [
        {
          id: 1,
          albumId: 1,
          title: "Test Photo",
          url: "/test.png",
          thumbnailUrl: "/test-thumbnail.png",
        },
        {
          id: 2,
          albumId: 1,
          title: "Test Photo2",
          url: "/test-2.png",
          thumbnailUrl: "/test-thumbnail-2.png",
        },
      ])
      .get("https://jsonplaceholder.typicode.com/users/1", {
        id: 1,
        name: "Adam",
        username: "adam1980",
        email: "adam@hoge.org",
        address: {
          street: "Kulas Light",
          suite: "Apt. 556",
          city: "Gwenborough",
          zipcode: "92998-3874",
          geo: {
            lat: "-37.3159",
            lng: "81.1496",
          },
        },
        phone: "1-770-736-8031 x56442",
        website: "adam.org",
        company: {
          name: "Test-Company",
          catchPhrase: "Next generation platform",
          bs: "photo",
        },
      });
  });

  test("should render correctly", async () => {
    const { page } = await getPage({
      route: "/album/1/",
    });

    const tree = renderer.create(page).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
