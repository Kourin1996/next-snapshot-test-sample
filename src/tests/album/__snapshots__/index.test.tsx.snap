// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album page should render correctly 1`] = `
<div
  className="container"
>
  <h1>
    Test Title
  </h1>
  <div
    className="album"
  >
    <div
      className="photos"
    >
      <div
        className="photo"
      >
        <img
          alt="Test Photo"
          src="/test.png"
        />
        <p>
          Test Photo
        </p>
      </div>
      <div
        className="photo"
      >
        <img
          alt="Test Photo2"
          src="/test-2.png"
        />
        <p>
          Test Photo2
        </p>
      </div>
    </div>
    <div
      className="album-author"
    >
      <div
        className="user-card"
      >
        <p
          className="user-card__name"
        >
          Adam
        </p>
        <p
          className="user-card__email"
        >
          adam@hoge.org
        </p>
        <p
          className="user-card__phone"
        >
          1-770-736-8031 x56442
        </p>
        <p
          className="user-card__website"
        >
          adam.org
        </p>
        <div
          className="user-card__company"
        >
          <p
            className="user-card__company__title"
          >
            Company Info
          </p>
          <p
            className="user-card__company__name"
          >
            Test-Company
          </p>
          <p
            className="user-card__company__catch-phrase"
          >
            Next generation platform
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
